### ISSUE TITLE
can no longer open on MacOS

### ISSUE THREAD

**Issue by jamestwebber** [igvteam/igv#1387]
This is a very confusing bug that I don't think is coming from IGV but I'm hoping you might have encountered it before.

As of today a fresh download of IGV will no longer open on my Mac, with an error that the application "may be damaged or incomplete". I checked with two other people and they had the same issue, but a person with a Windows machine downloaded it and was fine.

I am _guessing_ this is some permissions or security change in MacOS but I don't know how to fix it. Have you seen this before?

**Comment by helgathorv** [comment]
Is SentinelOne installed on your Mac? The latest version(s) of SentinelOne seem to think the IGV apps are malware. We're hoping to find a a solution for this, but for now a workaround is to download and run the command line version of IGV from the Downloads page. You'll need to have Java 11 installed on your Mac. .

**Comment by jamestwebber** [comment]
Gross! Okay I will try that, thanks for the quick response

**Comment by davidbernick** [comment]
Just as an important note -- while working with S1, we can send them a hash in advance via API. There are a number of ways to do this -- here's an example with a Slack extension. Can the IGV team help us with this so that each time a new version comes out, we'll make sure it gets Allowlisted WITH YOU. I'll post this to the GH issue, too. https://extensions.rapid7.com/extension/Blacklist-Hash-with-SentinelOne-from-Slack (among many other ways). Maybe a GH Action on publishing? Something like that. Sorry this is an issue -- S1 is THE primary way we prevent possible Ransomware and usually it's very good about this sorta stuff but for some reason (and we aren't getting good response from S1 surprisingly on this) it keeps flagging, specifically IGV.

**Comment by rickymagner** [comment]
It's working again for me today whereas yesterday I couldn't open it. 

IGV version: 2.16.0
MacOS Version: Ventura 13.4.1
Processor (since some originally suggested it might be related): Intel 2.4GHz i9

**Comment by jrobinso** [comment]
@davidbernick Just FYI the recent flurry of problems at the Broad was not due to an IGV release,  we haven't updated IGV since February.   It appears this was triggered by a S1 update.    Our builds are created on servers at the Broad,  not with GH actions. What exactly do you need from us?  You can use private email to myself or Louis Bergelson, we both have Broad emails, if you wish.

We're trying to figure out what triggers this in the first place as it effects many organizations using S1, not just the Broad.

**Comment by jamestwebber** [comment]
Thanks for the prompt attention to this, I got BITS involved as well and they may have reached out.

I'll add (particularly for users reading this) that my colleague Brian gave me a workaround: download the source distribution and you can run from the command-line with `igv.sh` (requires a Java runtime installed)

**Comment by jrobinso** [comment]
@jamestwebber The source distribution will work, but that's not required, the pre-built "Command Line ..." version should also work.  That can be downloaded here:  https://software.broadinstitute.org/software/igv/download

**Comment by jamestwebber** [comment]
Oh yeah that's what I meant.

**Comment by jrobinso** [comment]
@jamestwebber I would like to rename this issue to "Sentinel One blocks IGV packaged app on macOS" and keep it open until we find a permanent solution.

**Comment by ricktearle** [comment]
Has this issue been resolved?

**Comment by jrobinso** [comment]
@ricktearle There's no resolution with current releases other than whitelisting IGV.   We have no control over Sentinel One.

As mentioned above, the command line version of IGV does not trigger Sentinel One, as far as we know.

The next major release, 2.19,  uses a different technique to launch IGV on MacOS that may avoid Sentinel One triggers.   It seems to work with our limited testing so far.  This should be out by the end of the month.

**Comment by jrobinso** [comment]
If any of you having troubles with Sentinel One are up to a test could you try our latest patch release, 2.18.4?  One of the problems here is we really don't know for sure what exactly is triggering Sentinel One.   So please give this a try and let us know.  You can download it here:  https://igv.org/doc/desktop/#DownloadPage/

**Comment by ricktearle** [comment]
I downloaded 2.18.4 (with java) successfully, but the app does not launch. Interestingly, when I run get info, the date of last opened is current ie the app tried to launch, registered as launching, but nothing else happened. I am running Sequoia 15.0 on a M1 Max. I checked S1 and unlike 2.18.2, it did not trigger a response.

**Comment by jrobinso** [comment]
Well,  that's bad news and good news I guess.  We don't have that particular platform to test on.    I think your only option at the moment, other than whitelisting IGV by some means, is to run the "command line" version.  You just run a simple shell script from the command line to start.

**Comment by jrobinso** [comment]
@ricktearle I assume your machine is running Rosetta, correct?

**Comment by ricktearle** [comment]
Yes, running Rosetta, although I note that when I get info on 2.18.4, Rosetta is not an option.
Regardless, I can use the web browser version, but ultimately, we do need a solution.
It must be a bit deflating for you to make all this effort and not having IGV run out of the box.

**Comment by jrobinso** [comment]
Hi @ricktearle We've tried now on multiple Macs M1 with Sequoia and can't reproduce the issue.     However as I've suggested above you always have the option to run the generic ("Command Line IGV") version which you can download from here:  https://igv.org/doc/desktop/#DownloadPage/.   You need to install Java independently.  The instructions are in a readme.txt file when you unzip the archive, but I repeat them here.

```
1. Download and unzip the distribution file to a directory of your choice.

2. To start IGV execute one of the following launcher scripts from the command line:

igv-launcher.bat  (for Windows)
igv.bat           (for Windows batch jobs)
igv.sh            (for Linux and macOS)
igv_hidpi.sh      (for Linux with HiDPI displays)
igv.command       (macOS)

Some of these may not be present depending on the distribution you downloaded.  You might have to
make the script executable (e.g. chmod a+x igv.sh) depending on the way the bundle was unpacked.

```

**Comment by jrobinso** [comment]
So in summary there is a potential solution for the Sentinel One problem,  but it does not work on at least some Macs.  We will be rolling this back for now.  Whitelisting or the command line IGV are the workarounds.
